### PR TITLE
Rename ElementHandler -> ClassInformer

### PR DESCRIPTION
**Pull request by ZacSweers**

### CODE REVIEW

File: kotlinpoet-metadata-specs/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/ClassInspector.kt [hunk 1]
```diff
@@ -20,14 +20,12 @@ import com.squareup.kotlinpoet.metadata.ImmutableKmClass
 import com.squareup.kotlinpoet.metadata.KotlinPoetMetadataPreview
 import kotlinx.metadata.jvm.JvmMethodSignature
 
-/**
- * A basic interface for looking up information about JVM elements.
- */
+/** A basic interface for looking up information about JVM information about a given Class. */
```
Comment (Egorand) [inline kotlinpoet-metadata-specs/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/ClassInspector.kt]:
```suggestion
/** A basic interface for looking up JVM information about a given Class. */
```

File: kotlinpoet-metadata-specs/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/ClassInspector.kt [hunk 2]
```diff
@@ -20,14 +20,12 @@ import com.squareup.kotlinpoet.metadata.ImmutableKmClass
...
 @KotlinPoetMetadataPreview
-interface ElementHandler {
+interface ClassInspector {
 
   /**
-   * Indicates if this element handler supports [AnnotationRetention.RUNTIME]-retained annotations.
+   * Indicates if this informer supports [AnnotationRetention.RUNTIME]-retained annotations.
```
Comment (Egorand) [inline kotlinpoet-metadata-specs/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/ClassInspector.kt]:
```suggestion
   * Indicates if this [ClassInspector] supports [AnnotationRetention.RUNTIME]-retained annotations.
```

File: kotlinpoet-metadata-specs/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/KotlinPoetMetadataSpecs.kt [hunk 3]
```diff
@@ -271,11 +271,11 @@ private fun ImmutableKmClass.toTypeSpec(
       builder.addModifiers(INNER)
     }
     builder.addTypeVariables(typeParameters.map { it.toTypeVariableName(classTypeParamsResolver) })
-    // If we have an element handler, we can check exactly which "supertype" is an interface vs
+    // If we have an informer, we can check exactly which "supertype" is an interface vs
```
Comment (Egorand) [inline kotlinpoet-metadata-specs/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/KotlinPoetMetadataSpecs.kt]:
```suggestion
    // If we have an inspector, we can check exactly which "supertype" is an interface vs
```

**Comment by ZacSweers** [comment]
Updated with the suggestions 👍. Good catch

**Pull request merged**